server/game_components/game_objects.py

```diff
@@ -240,87 +240,68 @@ def action(self, _caster: Entity, _target: Entity | None) -> list[ActionDict]:
+        if cast:
             _caster.mana -= self.__cost
-            cast = True
 
         if _target is None:
-            if self.area_of_effect:
-                action_list: list[ActionDict] = []
-                targets: list[Mob | Player] = []
-                assert _caster.in_room
-                for mob in _caster.in_room.get_mobs():
-                    targets.append(mob)
-                for player in _caster.in_room.get_players():
-                    if player.uid != _caster.uid:
-                        targets.append(player)
-                for entity in targets:
-                    dmg = random.randint(self.__min_damage, self.__max_damage)
-                    if cast:
-                        hit_check = random.randint(0, 100)
-                        if hit_check <= self.__hit_percentage:
-                            hit = True
-                        if hit:
-                            entity.health -= dmg
-                        action_list.append(
-                            {
-                                "caster": _caster.uid,
-                                "target": entity.uid,
-                                "hit": hit,
-                                "dmg": dmg,
-                                "cast": cast,
-                            }
-                        )
-                    else:
-                        action_list.append(
-                            {
-                                "caster": _caster.uid,
-                                "target": entity.uid,
-                                "hit": hit,
-                                "dmg": dmg,
-                                "cast": cast,
-                            }
-                        )
-                return action_list
-            else:
-                _target = _caster
-
-        if not self.area_of_effect:
-            dmg = random.randint(self.__min_damage, self.__max_damage)
-            if cast:
-                hit_check = random.randint(0, 100)
-                if hit_check <= self.__hit_percentage:
-                    hit = True
-                if hit:
-                    _target.health -= dmg
-
-                return [
-                    {
-                        "caster": _caster.uid,
-                        "target": _target.uid,
-                        "hit": hit,
-                        "dmg": dmg,
-                        "cast": cast,
-                    }
-                ]
-            else:
-                return [
-                    {
-                        "caster": _caster.uid,
-                        "target": _target.uid,
-                        "hit": hit,
-                        "dmg": dmg,
-                        "cast": cast,
-                    }
-                ]
+            assert self.requires_target is False
+            action_list = self._no_target_action(cast, _caster)
+        else:
+            assert self.requires_target is True
```

Comment:
this could be shortened to `assert self.requires_target`, on top of my other comments above.